### PR TITLE
fix: added support for multiple form elements on one page

### LINKED ISSUE #50: Multiple form elements on one page all display thank you message after one is submitted
The presence of the /finished/#uff in the URL makes all UDF elements on the same page display the Thank you message even though only one was submitted.

## PRs
- https://github.com/dnadesign/silverstripe-elemental-userforms/pull/83

### PR DESCRIPTION
**Pull request by rasstislav**
fixed #50

### CODE REVIEW

File: src/Model/ElementForm.php [hunk 1]
```diff
@@ -44,6 +47,11 @@ public function Form()
         $link = $current->Link();
         restore_error_handler();
 
+        if ($isFinishedAction) {
+            $pattern = '#element/\d+#'; // Regular expression pattern for handleElement (element/$ID!)
+            $link = preg_replace($pattern, '', $link);
+        }
```
Comment (ChloeSartorelli) [inline src/Model/ElementForm.php]: While this works in this specific scenario, it feels a bit flimsy relying on hardcoded URL segments.
This could fall apart if, for example, the elemental userform is on a page (probably with a custom controller) which has the url `/some-page/element/1/` - the link for the userform element then would end up being `/some-page/element/1/element/<id>/` and the regex would match and remove both sets of element and ID.

One solution would be to make the following change in `ElementFormController`:
```diff
 public function Link($action = null)
 {
-    $id = $this->element->ID;
-    $segment = Controller::join_links('element', $id, $action);
     $page = Director::get_current_page();
+    $id = $this->element->ID;
+    if ($this->getAction() === 'finished'
+        && $action !== 'finished'
+        && $this->getRequest()->param('ID') == $id
+    ) {
+        $segment = $action;
+    } else {
+        $segment = Controller::join_links('element', $id, $action);
+    }
 
     if ($page && !($page instanceof ElementController)) {
```

This avoids adding the `element/<id>/` portion of the URL if:
1. We are handling a "finished" request
2. The URL we're fetching is not for an action called "finished" (could also swap this out for `($action !== '' && $action !== null)`)
3. The ID the "finished" request is for is the current element

I feel like even that might have some edge cases I've not considered though...

Comment (rasstislav) [reply src/Model/ElementForm.php]: This solution is much better than mine. I implemented the changes.
Thanks :-)

**Review by GuySartorelli — APPROVED** [review]
I was going to ask for some behat test coverage for this scenario, but I took a look and the current behat tests don't test the front-end _at all_ so I think that would be out of scope for this PR.

I've been thinking this over, as I mentioned in my previous change request I wasn't sure if there were edge cases that this would break. I can't think of any specific scenario where that would happen - I don't think it's likely at all that someone has customisations for this elemental block, rather customisations are likely to be in the userforms itself which this change won't affect.

Thank you for submitting this. It works great.
